docs/07-Contributing/developing.md

```diff
@@ -190,8 +190,25 @@ In order to certify the provenance of commits and defend against impersonation,
 Documentation for setting up Git and Github to sign your commits can be found [here](https://docs.github.com/en/authentication/managing-commit-signature-verification/signing-commits).
 Additional information about Git's use of GPG can be found [here](https://git-scm.com/book/en/v2/Git-Tools-Signing-Your-Work)
 
+This can be accomplished by providing a `-S` flag to `git commit` as documented [here](https://git-scm.com/docs/git-commit#Documentation/git-commit.txt--Sltkeyidgt)
```

Comment:
You can also do `git config commit.gpgSign true` and forget about the flag. You probably don't want `--global` in this instance (unless you really want to sign everything, I do not).

Reply:
@timraymond I agree and I have that set too in my local gitconfig. However, this is literally the first tip of the [Signing Commits doc](https://docs.github.com/en/authentication/managing-commit-signature-verification/signing-commits) which is already shared in this section.

The reason why I added that sentence is because we constantly have commits that do not have signature verification and this is causing unnecessary engagement from contributors/maintainers.

I rather incentivize contributors to read the GitHub docs from the provided link (and use `git config commit.gpgSign true`), or at the very least read this sentence I just added and use the `-S` flag.

Reply:
@SRodi The docs are not perfect. Providing `-S` is an administrative control that will eventually be forgotten by someone, requiring continued muda circling back with contributors. Telling contributors to modify their git config for this repo is an engineering control that forever solves this problem for a particular contributor. Honestly, I wouldn't be opposed to a `Makefile` target that just does it for you.

I'm not suggesting we change your existing sentence, but I'm saying we should add a recommendation to set `commit.gpgSign` for this repo.

Reply:
maybe in here, as a default for the codespaces users
https://github.com/microsoft/retina/blob/main/.devcontainer/installMoreTools.sh

Reply:
thanks for the comments! I have updated this PR to include a check in GH workflow to verify GPG signature exist, updated the doc to explicitly suggest using `git config commit.gpgSign true` and added the same instruction to codespaces setup (installMOreTools.sh) as mentioned above.